_analyzers/token-filters/common_gram.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: Common grams
+parent: Token filters
+nav_order: 60
+---
+<!-- vale off -->
+# Common grams token filter
+<!-- vale on -->
+The `common_grams` token filter in OpenSearch improves search relevance by keeping commonly occurring phrases (common grams) in the text. This is useful when dealing with languages or datasets where certain word combinations frequently occur as a unit and can impact the search relevance if treated as separate tokens. If any common words are present in the input string, this token filter generates both their unigrams and bigrams.
+
+Using this token filter improves search relevance by keeping common phrases intact. This can help in matching queries more accurately, particularly for frequent word combinations. It also improves search precision by reducing the number of irrelevant matches.
+
+When using this filter, you must carefully select and maintain the `common_words` list.
+{: .warning}
+
+## Parameters
+
+The `common_grams` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`common_words` | Required | List of strings | A list of words that should be considered as words appearing together. These words will be used to generate common grams. If the `common_words` parameter is an empty list, the `common_grams` token filter becomes a no-op filter, meaning it doesn't modify the input tokens at all.
+`ignore_case` | Optional | Boolean |  Indicates whether the filter should ignore case differences when matching common words. Default is `false`.
```

Comment:
Can we make "treated as words appearing together" a bit more precise? Appearing together how/in what way?

Reply:
Commonly appearing together? This is physically together, like in a phrase.

Reply:
"treated as words that commonly appear together"?